iocore/dns/DNS.cc

```diff
@@ -1553,32 +1553,56 @@ dns_process(DNSHandler *handler, HostEnt *buf, int len)
     goto Lerror;
   }
 
+  // Logs using SiteThrottled* version is helpful for noisy logs, being used here
+  // instead of print statements to help with the possible retries when a dns code occurs
   if (h->rcode != NOERROR || !h->ancount) {
     Debug("dns", "received rcode = %d", h->rcode);
     switch (h->rcode) {
     default:
-      Warning("Unknown DNS error %d for [%s]", h->rcode, e->qname);
+      SiteThrottledWarning("Unknown: DNS error %d for [%s]", h->rcode, e->qname);
       retry     = true;
       server_ok = false; // could be server problems
       goto Lerror;
-    case SERVFAIL: // recoverable error
+    case NOERROR: // NOERROR the if condition excludes this status code to avoid spamming the logs, good for testing
+      Debug("dns", "NOERROR: DNS code %d for [%s]: DNS Query completed successfully.", h->rcode, e->qname);
+      break;
+    case FORMERR: // FORMERR unrecoverable errors
+      SiteThrottledNote("FORMERR: DNS error %d for [%s]: DNS Query Format Error.", h->rcode, e->qname);
+      break;
```

Comment:
`FORMERR` used to `goto Lerror`, now it breaks. I think since it is unrecoverable, we still want to `goto Lerror` right?

Reply:
please refer to my comment below.